src/grid_interpolator.cc

```diff
@@ -0,0 +1,99 @@
+#include "flyft/grid_interpolator.h"
+
+#include <fstream> 
+#include <cassert>
+
+namespace flyft
+{
+GridInterpolator::GridInterpolator():
+    n_(0,0,0)
+    {
+    }
+    
+GridInterpolator::GridInterpolator(const std::string& s)
+    {
+    int nx, ny, nz;
+    double dx, dy, dz;
+    
+    std::ifstream indata(s);
+    if(!indata)
+        {
+        throw std::runtime_error("File failed to open");
+        }
+        
+    indata>>nx>>ny>>nz>>dx>>dy>>dz;
+    n_ = ThreeDimensionIndex(nx,ny,nz);
+    
+    if (n_.size() == 0)
+        {
+        throw std::runtime_error("Grid has zero size");
+        }
+        
+    data_ = new double[n_.size()];  
+    double value;
+    
+    int idx = 0;
+    while (indata >> value)
+        {
+        data_[idx++] = value;
+        }
+        
+    if (idx != n_.size() || !indata.eof())
+        {
+        throw std::runtime_error("File did not right number of elements");
+        }
+        
+    indata.close();
+    }
+    
+   
+
+GridInterpolator::~GridInterpolator()
+    {
+    delete[] data_;
+    }
+    
+
+double GridInterpolator::operator()(double x, double y, double z) const
+    {
+    
+    int bin_x = std::floor(x/dx_); 
+    int bin_y = std::floor(y/dy_); 
+    int bin_z = std::floor(z/dz_); 
+    int ni, nj , nk;
+    std::tie(ni, nj, nk) = n_.getBounds();
+    assert(bin_x >= 0 && bin_x < ni);
+    assert(bin_y >= 0 && bin_y < nj);
+    assert(bin_z >= 0 && bin_z < nk);
```

Comment:
If this is the only spot it's used, can you just take this in the tuple and then use get on it?
```suggestion
    const auto bounds = n_.getBounds();
    assert(bin_x >= 0 && bin_x < std::get<0>(bounds));
    assert(bin_y >= 0 && bin_y < std::get<1>(bounds));
    assert(bin_z >= 0 && bin_z < std::get<2>(bounds));
```